src/transformers/modeling_utils.py

```diff
@@ -4047,6 +4048,14 @@ def from_pretrained(
                             )
                     elif hasattr(torch, torch_dtype):
                         torch_dtype = getattr(torch, torch_dtype)
+                        for sub_config_key in config.sub_configs.keys():
+                            sub_config = getattr(config, sub_config_key)
+                            sub_config.torch_dtype = torch_dtype
+                    elif isinstance(torch_dtype, dict):
+                        for key, curr_dtype in torch_dtype.items():
+                            if hasattr(config, key):
+                                value = getattr(config, key)
+                                value.torch_dtype = curr_dtype
```

Comment:
if users passes one torch dtype as before, we just use it in all sub-configs. Otherwise a user can either set directly dtypes in configs before loading the model, or indicate a dict `torch_dtype` when loading similarly to `attn_implementation_dispatch`